idaes/models_extra/gas_solid_contactors/unit_models/moving_bed.py

```diff
@@ -433,12 +498,49 @@ def build(self):
         # Get units meta data from property packages
         units_meta_solid = solid_phase.property_package.get_metadata().get_derived_units
 
-        # Create a unit model length domain
-        self.length_domain = ContinuousSet(
-            bounds=(0.0, 1.0),
-            initialize=self.config.length_domain_set,
-            doc="Normalized length domain",
-        )
+        # In this branch, we assume that either both gas and solid
+        # transformation schemes are specified, or neither are specified.
+        if (
+            # solid and gas transformation schemes are specified
+            (self.config.solid_transformation_scheme is not None)
+            and (self.config.gas_transformation_scheme is not None)
+        ):
+            # Create length domain sets for gas and solid phases. This is
+            # necessary so we can apply different discretizations to the
+            # different phases.
+            self._gas_tr_scheme = self.config.gas_transformation_scheme
+            self._solid_tr_scheme = self.config.solid_transformation_scheme
+            self.solid_length_domain = ContinuousSet(
+                bounds=(0.0, 1.0),
+                initialize=self.config.length_domain_set,
+                doc="Solid phase normalized length domain",
+            )
+            self.gas_length_domain = ContinuousSet(
+                bounds=(0.0, 1.0),
+                initialize=self.config.length_domain_set,
+                doc="Gas phase normalized length domain",
+            )
+            # Create an instance attribute "length_domain" for convenience.
+            # Arbitrarily, use the solid phase length domain.
+            #
+            # We do not use Reference as it does not work for Sets.
+            super(_BlockData, self).__setattr__(
```

Comment:
Minor point, but we would normally use `add_object_reference` for this (as this is what it does behind the scenes).